mysql-test/suite/innodb/t/foreign_null.test

```diff
@@ -0,0 +1,110 @@
+--source include/have_innodb.inc
+let $algorithm=`select regexp_replace('$MTR_COMBINATIONS', 'innodb,\|,innodb', '')`;
+let $error_code = 0;
+
+if ($algorithm == "COPY") {
+ let $error_code = ER_FK_COLUMN_CANNOT_CHANGE;
+}
+
+if ($algorithm == "INPLACE") {
+ let $error_code = ER_FK_COLUMN_NOT_NULL;
+}
+
+# MODIFY CHILD COLUMN..NOT NULL ON UPDATE CASCADE.. PARENT COLUMN NULL
+CREATE TABLE t1(f1 INT, f2 INT, PRIMARY KEY(f1), KEY(f2))ENGINE=InnoDB;
+CREATE TABLE t2(f1 INT, FOREIGN KEY(f1) REFERENCES t1(f2) ON UPDATE CASCADE)ENGINE=InnoDB;
+replace_result ,ALGORITHM=COPY '' ,ALGORITHM=INPLACE '';
+--error $error_code
+eval ALTER TABLE t2 MODIFY COLUMN f1 INT NOT NULL,ALGORITHM=$algorithm;
```

Comment:
1. The comment is actually useful in result file. I would also suggest changing NULL to NULLABLE or NO CONSTRAINT for parent table in the comment.
`--echo Test-1: MODIFY CHILD COLUMN..NOT NULL ON UPDATE CASCADE.. PARENT COLUMN NULLABLE`
2. Can we add a case for ON DELETE CASCADE to make sure we are allowing it to pass ?